src/json-rpc-provider-types.ts

```diff
@@ -0,0 +1,95 @@
+import type SafeEventEmitter from '@metamask/safe-event-emitter';
+
+import type { JsonRpcParams, JsonRpcRequest, Json } from './json';
+import type { PartialOrAbsent } from './misc';
+
+/**
+ * An interface for the EIP-1193 specification for an Ethereum JavaScript Provider.
+ *
+ * For details, see:
+ * - [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193)
+ * - [BaseProvider]{@link https://github.com/MetaMask/providers/blob/main/src/BaseProvider.ts} in package [@metamask/providers](https://www.npmjs.com/package/@metamask/providers)
+ */
+export type EIP1993Provider = SafeEventEmitter & {
+  /**
+   * Submits an RPC request for the given method, with the given params.
+   * Resolves with the result of the method call, or rejects on error.
+   *
+   * @param args - The RPC request arguments.
+   * @param args.method - The RPC method name.
+   * @param args.params - The parameters for the RPC method.
+   * @returns A Promise that resolves with the result of the RPC method,
+   * or rejects if an error is encountered.
+   */
+  request<Params extends JsonRpcParams, Result extends Json>(
+    args: Params,
+  ): Promise<PartialOrAbsent<Result>>;
+};
+
+/**
+ * The interface for a legacy Ethereum provider.
+ *
+ * This provider follows conventions that pre-date
+ * [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193). It is not compliant with
+ * any Ethereum provider standard.
+ */
+export type LegacyEthereumProvider =
+  | LegacyEthersProvider
+  | LegacyEthJsQueryProvider
+  | LegacyWeb3Provider;
+
+type LegacyEthersProvider = {
+  /**
+   * Send a provider request asynchronously. (ethers v5 Web3Provider)
+   *
+   * @param method - The RPC method to call.
+   * @param params - Array with method parameters.
+   * @returns A promise resolving with the result of the RPC call, or rejecting on failure.
+   */
+  send(method: string, params: any[]): Promise<Json>;
```

Comment:
Could we type params with `unknown[]` or `string[]`?
```suggestion
send(method: string, params: unknown[]): Promise<Json>;
```

Reply:
I think we should still use `unknown` here. The range of types that can be assigned to `params` is exactly the same whether we use `any[]` or `unknown[]`. Any errors resulting from `unknown` will also cause `any` to fail -- just at runtime and silently. 

We can always use `@ts-expect-error` if the error block something. At least then we can be aware that there's a dragon to account for.